samples/performance/bindless/bindless.cpp

```diff
@@ -0,0 +1,925 @@
+				break;
+		}
+
+		vkUpdateDescriptorSets(get_device().get_handle(), static_cast<uint32_t>(write_descriptor_sets.size()), write_descriptor_sets.data(), 0, VK_NULL_HANDLE);
+	};
+
+	// render pipeline
+	bind(descriptor_set, descriptor_set_layout, Target::RenderPipeline);
+
+	//  compute pipeline
+	bind(gpu_cull_descriptor_set, gpu_cull_descriptor_set_layout, Target::ComputePipeline);
+
+	// Device address pipeline
+	if (m_supports_buffer_device)
+	{
+		bind(device_address_descriptor_set, device_address_descriptor_set_layout, Target::AddressPipeline);
+	}
+}
+
+void BindlessResources::update_scene_uniform()
+{
+	if (!scene_uniform_buffer)
+	{
+		scene_uniform_buffer = std::make_unique<vkb::core::Buffer>(*device, sizeof(SceneUniform), VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT, VMA_MEMORY_USAGE_CPU_TO_GPU);
+	}
+	scene_uniform.proj = camera.matrices.perspective;
+	scene_uniform.view = camera.matrices.view;
+
+	scene_uniform_buffer->update(&scene_uniform, sizeof(scene_uniform), 0);
+}
+
+void BindlessResources::draw()
+{
+	ApiVulkanSample::prepare_frame();
+
+	// Command buffer to be submitted to the queue
+	submit_info.commandBufferCount = 1;
+	submit_info.pCommandBuffers    = &draw_cmd_buffers[current_buffer];
+
+	// Submit to queue
+	VK_CHECK(vkQueueSubmit(queue, 1, &submit_info, VK_NULL_HANDLE));
+
+	ApiVulkanSample::submit_frame();
+}
+
+void BindlessResources::render(float delta_time)
+{
+	if (!prepared)
+	{
+		return;
+	}
+
+	if (render_mode == GPU_DEVICE_ADDRESS && !m_supports_buffer_device)
+	{
+		render_mode = GPU;
+	}
+
+	if (m_requires_rebuild)
+	{
+		build_command_buffers();
+		m_requires_rebuild = false;
+	}
+
+	draw();
+
+	update_scene_uniform();
+
+	run_cull();
+	device->get_fence_pool().wait();
+	device->get_fence_pool().reset();
+}
+
+void BindlessResources::finish()
+{
+}
+
+void BindlessResources::run_cull()
+{
+	switch (render_mode)
+	{
+		case RenderMode::CPU:
+			cpu_cull();
+			break;
+		case RenderMode::GPU:
+		case RenderMode::GPU_DEVICE_ADDRESS:
+			run_gpu_cull();
+			break;
+	}
+}
+
+void BindlessResources::run_gpu_cull()
+{
+	assert(!!gpu_cull_pipeline);
+	auto                     cmd   = device->create_command_buffer(VK_COMMAND_BUFFER_LEVEL_PRIMARY);
+	VkCommandBufferBeginInfo begin = vkb::initializers::command_buffer_begin_info();
+
+	vkBeginCommandBuffer(cmd, &begin);
+	auto bind = [&cmd](VkPipeline &_pipeline, VkPipelineLayout &_pipeline_layout, VkDescriptorSet &_descriptor_set) {
+		vkCmdBindPipeline(cmd, VK_PIPELINE_BIND_POINT_COMPUTE, _pipeline);
+		vkCmdBindDescriptorSets(cmd, VK_PIPELINE_BIND_POINT_COMPUTE, _pipeline_layout, 0, 1, &_descriptor_set, 0, nullptr);
+	};
+
+	if (render_mode == RenderMode::GPU)
+	{
+		bind(gpu_cull_pipeline, gpu_cull_pipeline_layout, gpu_cull_descriptor_set);
+	}
+	else
+	{
+		bind(device_address_pipeline, device_address_pipeline_layout, device_address_descriptor_set);
+	}
+
+	vkCmdDispatch(cmd, models.size(), 1, 1);
```

Comment:
This would effectively dispatch N work-groups consisting of 1 thread, would it? Would it make more sense, performance wise, to dispatch e.g. N/64 workgroups of 64 threads -- so 64-wide warps can be fully filled?